auditbeat/module/file_integrity/monitor/recursive.go

```diff
@@ -84,37 +84,60 @@ func (watcher *recursiveWatcher) ErrorChannel() <-chan error {
 	return watcher.inner.Errors
 }
 
+func (watcher *recursiveWatcher) watchFile(path string, info os.FileInfo) error {
+	var err error
+	if info == nil {
+		info, err = os.Lstat(path)
+		if err != nil {
+			return err
+		}
+	}
+
+	if info.IsDir() {
+		if err = watcher.tree.AddDir(path); err == nil {
+			if err = watcher.inner.Add(path); err != nil {
+				return fmt.Errorf("failed adding watcher to '%s': %w", path, err)
```

Comment:
```suggestion
				return err
```
otherwise this is going to wrap the error msg twice for the same path, see L114, L137

maybe you'd like to add a specific message here, for example "failed adding inner watcher..."?

Reply:
> otherwise this is going to wrap the error msg twice for the same path, see L114, L137

yes the wrapping here is unnecessary

> anyway, since the paths errors out, don't you want to remove the watcher on the Dir ?

@intxgo based on the original code on the left (this code used to be part of addRecursive and now it is detached from there as a func named `watchFile` but still called from there) there was no removal on err so no removal on err now. Specifically, a watcher is added only for directory entries not for files, and according to initial design if adding a watcher fails you are not expected to remove the dir from the tree. This design choice is captured by the unittest `TestRecursiveSubdirPermissions` as this fails if you remove the dir on adding the watcher error. Also if there is an error here the idea is that the whole watcher exits.

Reply:
LGTM then, thank